### PR TITLE
Check whether outgoing message is in a loop before sending

### PR DESCRIPTION
**Pull request by nicpottier**
This basically means that any time a contact gets the same flow message more than 20 times in a 5 minute window we will fail it. I *think* that's reasonable.

### CODE REVIEW

File: backends/rapidpro/backend.go [hunk 1]
```diff
@@ -183,6 +183,63 @@ func (b *backend) WasMsgSent(ctx context.Context, msg courier.Msg) (bool, error)
...
+			count = tonumber(record_count) + 1
+		else
+			count = 1
+		end		
+	end
+
+	-- create our new record with our updated count
+	record = string.format("%02d:%s", count, text)
+
+	-- write our new record witn updated count
```
Comment (rowanseymour) [inline backends/rapidpro/backend.go]: typo

File: backends/rapidpro/backend.go [hunk 2]
```diff
@@ -183,6 +183,63 @@ func (b *backend) WasMsgSent(ctx context.Context, msg courier.Msg) (bool, error)
...
+	defer rc.Close()
+
+	keyTime := time.Now().UTC().Round(time.Minute * 5)
+	key := fmt.Sprintf(sentSetName, fmt.Sprintf("loop_msgs:%s", keyTime.Format("2006-01-02-15:04")))
+	count, err := redis.Int(luaMsgLoop.Do(rc, key, m.ContactID_, m.Text_))
+	if err != nil {
+		return false, errors.Wrapf(err, "error while checking for msg loop")
+	}
+
+	if count >= 20 {
```
Comment (rowanseymour) [inline backends/rapidpro/backend.go]: const ?

File: sender.go [hunk 3]
```diff
@@ -177,21 +177,34 @@ func (w *Sender) sendMessage(msg Msg) {
...
-		msgLog.WithError(err).Warning("error looking up msg was sent")
+		log.WithError(err).Warning("error looking up msg was sent")
+	}
+
+	// is this msg in a loop?
+	loop, err := backend.IsMsgLoop(sendCTX, msg)
+
+	// failing on loop lookup isn't permanent, but log
+	if err != nil {
+		log.WithError(err).Warning("error looking up msg loop")
```
Comment (rowanseymour) [inline sender.go]: if this is just a warning, any chance it stops working and we never notice?

Comment (nicpottier) [reply sender.go]: I *think* warnings still show up in Sentry, but actually I do prefer these being real errors from a sentry perspective, will change.

File: backends/rapidpro/backend.go (hunk 2 above)
Comment (nicpottier) [reply backends/rapidpro/backend.go]: If something is only used once I feel like naming it into a const is just hiding it, but can change.

Comment (rowanseymour) [reply backends/rapidpro/backend.go]: that's fair - was just thinking from a "advertising that something is configurable" pov

**Pull request merged**